hibernate-graalvm/src/test/java/org/hibernate/graalvm/internal/StaticClassListsTest.java

```diff
@@ -37,6 +39,37 @@ public static void index() throws IOException {
 		hibernateIndex = JandexTestUtils.indexJar( Session.class );
 	}
 
+	@Nested
+	class TypesNeedingRuntimeInitialization {
+		@ParameterizedTest
+		@EnumSource(TypesNeedingRuntimeInitialization_Category.class)
+		void containsAllExpectedClasses(TypesNeedingRuntimeInitialization_Category category) {
+			assertThat( StaticClassLists.typesNeedingRuntimeInitialization() )
+					.containsAll( category.classes().collect( Collectors.toSet() ) );
+		}
+
+		@Test
+		void meta_noMissingTestCategory() {
+			assertThat( Arrays.stream( TypesNeedingRuntimeInitialization_Category.values() ).flatMap( TypesNeedingRuntimeInitialization_Category::classes ) )
+					.as( "If this fails, a category is missing in " + TypesNeedingRuntimeInitialization_Category.class )
+					.contains( StaticClassLists.typesNeedingRuntimeInitialization() );
+		}
+	}
+
+	enum TypesNeedingRuntimeInitialization_Category {
+		MISC {
```

Comment:
The intent was to have categories so that we know what each set of classes is about.
Maybe rename this to `UUID_STRATGY_HOLDERS_USING_SECURE_RANDOM`?
Alternatively, add a comment to `StaticClassLists.java` to explain that in native images, `SecureRandom` needs to be initialized at runtime, which precludes initialiazing `Holder` at build-time (since it creates a `SecureRandom` during its initialization)?